In [2]:
#importing Pandas
import pandas as pd

In [3]:
#reading Box Office Mojo database
bom_df = pd.read_csv("zippedData/bom.movie_gross.csv.gz", compression='gzip')

In [4]:
bom_df.head()

title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

  foreign_gross  year  
0     652000000  2010  
1     691300000  2010  
2     664300000  2010  
3     535700000  2010  
4     513900000  2010

In [5]:
bom_df.head(20)

title studio  domestic_gross  \
0                                         Toy Story 3     BV     415000000.0   
1                          Alice in Wonderland (2010)     BV     334200000.0   
2         Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                           Inception     WB     292600000.0   
4                                 Shrek Forever After   P/DW     238700000.0   
5                          The Twilight Saga: Eclipse   Sum.     300500000.0   
6                                          Iron Man 2   Par.     312400000.0   
7                                             Tangled     BV     200800000.0   
8                                       Despicable Me   Uni.     251500000.0   
9                            How to Train Your Dragon   P/DW     217600000.0   
10                         Clash of the Titans (2010)     WB     163200000.0   
11  The Chronicles of Narnia: The Voyage of the Da...    Fox     104400000.0   
12                                  The King's Speech  Wein.     135500000.0   
13                                        Tron Legacy     BV     172100000.0   
14                                     The Karate Kid   Sony     176600000.0   
15                Prince of Persia: The Sands of Time     BV      90800000.0   
16                                         Black Swan   FoxS     107000000.0   
17                                           Megamind   P/DW     148400000.0   
18                                         Robin Hood   Uni.     105300000.0   
19                                 The Last Airbender   Par.     131800000.0   

   foreign_gross  year  
0      652000000  2010  
1      691300000  2010  
2      664300000  2010  
3      535700000  2010  
4      513900000  2010  
5      398000000  2010  
6      311500000  2010  
7      391000000  2010  
8      291600000  2010  
9      277300000  2010  
10     330000000  2010  
11     311300000  2010  
12     275400000  2010  
13     228000000  2010  
14     182500000  2010  
15     245600000  2010  
16     222400000  2010  
17     173500000  2010  
18     216400000  2010  
19     187900000  2010

In [6]:
bom_df.tail(20)

title      studio  \
3367                      Elliot: The Littlest Reindeer       Scre.   
3368                                       The Workshop      Strand   
3369                                       Loving Pablo        Uni.   
3370                                  A Paris Education          KL   
3371                          Tag Along: The Devil Fish          CL   
3372                           Gabriel and the Mountain      Strand   
3373                  Marlina the Murderer in Four Acts       Icar.   
3374                                         The Escape         IFC   
3375  How Victor 'The Garlic' Took Alexey 'The Stud'...     CARUSEL   
3376        Lou Andreas-Salomé, the Audacity to be Free         CLS   
3377                                    Let Yourself Go        Men.   
3378                                      Hannah (2018)         PDF   
3379                                           Souvenir      Strand   
3380                       Furious (Legend of Kolovrat)     CARUSEL   
3381                                Beauty and the Dogs       Osci.   
3382                                          The Quake       Magn.   
3383                        Edward II (2018 re-release)          FM   
3384                                           El Pacto        Sony   
3385                                           The Swan  Synergetic   
3386                                  An Actor Prepares       Grav.   

      domestic_gross foreign_gross  year  
3367         24300.0           NaN  2018  
3368         22100.0           NaN  2018  
3369         22000.0           NaN  2018  
3370         21600.0           NaN  2018  
3371         20700.0           NaN  2018  
3372         18400.0           NaN  2018  
3373         17800.0           NaN  2018  
3374         14000.0           NaN  2018  
3375         13000.0           NaN  2018  
3376         12900.0           NaN  2018  
3377         12200.0           NaN  2018  
3378         11700.0           NaN  2018  
3379         11400.0           NaN  2018  
3380         10000.0           NaN  2018  
3381          8900.0           NaN  2018  
3382          6200.0           NaN  2018  
3383          4800.0           NaN  2018  
3384          2500.0           NaN  2018  
3385          2400.0           NaN  2018  
3386          1700.0           NaN  2018

In [7]:
bom_df.year.value_counts().sort_index()

2010    328
2011    399
2012    400
2013    350
2014    395
2015    450
2016    436
2017    321
2018    308
Name: year, dtype: int64

Have data for 2010-2018

In [8]:
bom_df.studio.value_counts().head(60)

IFC         166
Uni.        147
WB          140
Magn.       136
Fox         136
SPC         123
Sony        110
BV          106
LGF         103
Par.        101
Eros         89
Wein.        77
CL           74
Strand       68
FoxS         67
RAtt.        66
KL           62
Focus        60
WGUSA        58
CJ           56
MBox         54
UTV          50
A24          49
WB (NL)      45
FM           42
LG/S         41
Cohen        40
ORF          37
SGem         35
Rela.        35
FIP          26
GK           24
STX          24
Gold.        24
Osci.        23
TriS         23
RTWC         23
MNE          22
BST          22
EOne         21
Distrib.     20
Drft.        20
Relbig.      18
Anch.        18
CBS          17
BG           16
Zeit.        16
Sum.         15
FD           14
Yash         14
W/Dim.       14
Trib.        14
Orch.        13
Free         12
FCW          12
FUN          12
FRun         12
Elev.        11
Vari.        11
PNT          11
Name: studio, dtype: int64

In [9]:
bom_df.studio.isna().sum()

5

In [10]:
bom_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3387 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3387 non-null   object 
 1   studio          3382 non-null   object 
 2   domestic_gross  3359 non-null   float64
 3   foreign_gross   2037 non-null   object 
 4   year            3387 non-null   int64  
dtypes: float64(1), int64(1), object(3)
memory usage: 132.4+ KB


In [11]:
bom_df.loc[bom_df.year==2018]

title      studio  domestic_gross  \
3079          Avengers: Infinity War          BV     678800000.0   
3080                   Black Panther          BV     700100000.0   
3081  Jurassic World: Fallen Kingdom        Uni.     417700000.0   
3082                   Incredibles 2          BV     608600000.0   
3083                         Aquaman          WB     335100000.0   
...                              ...         ...             ...   
3382                       The Quake       Magn.          6200.0   
3383     Edward II (2018 re-release)          FM          4800.0   
3384                        El Pacto        Sony          2500.0   
3385                        The Swan  Synergetic          2400.0   
3386               An Actor Prepares       Grav.          1700.0   

     foreign_gross  year  
3079       1,369.5  2018  
3080     646900000  2018  
3081     891800000  2018  
3082     634200000  2018  
3083     812700000  2018  
...            ...   ...  
3382           NaN  2018  
3383           NaN  2018  
3384           NaN  2018  
3385           NaN  2018  
3386           NaN  2018  

[308 rows x 5 columns]

In [12]:
bom_df.domestic_gross.isna().sum()

28

In [13]:
bom_df.loc[bom_df.foreign_gross.isna() & bom_df.domestic_gross.isna()]

Empty DataFrame
Columns: [title, studio, domestic_gross, foreign_gross, year]
Index: []

Box Office Mojo

-All the rows have years and titles.  
-There are 5 missing studios--perhaps these are independent films? <br>
-the studios might have different names for the same studio. need to fix. <br>
-we have domestic and foreign gross.  could add together for total gross.<br>
-both domestic and foreign have nulls-probably only marketed domestically or non-domestically. none have nulls for both. could impute 0

In [14]:
#read Rotten Tomatoes Movie Info
rt_mi_df = pd.read_csv("zippedData/rt.movie_info.tsv.gz", sep = '\t', compression = 'gzip')

In [15]:
rt_mi_df.head()

id                                           synopsis rating  \
0   1  This gritty, fast-paced, and innovative police...      R   
1   3  New York City, not-too-distant-future: Eric Pa...      R   
2   5  Illeana Douglas delivers a superb performance ...      R   
3   6  Michael Douglas runs afoul of a treacherous su...      R   
4   7                                                NaN     NR   

                                 genre          director  \
0  Action and Adventure|Classics|Drama  William Friedkin   
1    Drama|Science Fiction and Fantasy  David Cronenberg   
2    Drama|Musical and Performing Arts    Allison Anders   
3           Drama|Mystery and Suspense    Barry Levinson   
4                        Drama|Romance    Rodney Bennett   

                            writer  theater_date      dvd_date currency  \
0                   Ernest Tidyman   Oct 9, 1971  Sep 25, 2001      NaN   
1     David Cronenberg|Don DeLillo  Aug 17, 2012   Jan 1, 2013        $   
2                   Allison Anders  Sep 13, 1996  Apr 18, 2000      NaN   
3  Paul Attanasio|Michael Crichton   Dec 9, 1994  Aug 27, 1997      NaN   
4                     Giles Cooper           NaN           NaN      NaN   

  box_office      runtime             studio  
0        NaN  104 minutes                NaN  
1    600,000  108 minutes  Entertainment One  
2        NaN  116 minutes                NaN  
3        NaN  128 minutes                NaN  
4        NaN  200 minutes                NaN

The Rotten Tomatoes Movie Info DataFrame contains info for select movies, but not the titles, so probably need to merge with another dataframe. 

In [16]:
rt_mi_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1560 entries, 0 to 1559
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   id            1560 non-null   int64 
 1   synopsis      1498 non-null   object
 2   rating        1557 non-null   object
 3   genre         1552 non-null   object
 4   director      1361 non-null   object
 5   writer        1111 non-null   object
 6   theater_date  1201 non-null   object
 7   dvd_date      1201 non-null   object
 8   currency      340 non-null    object
 9   box_office    340 non-null    object
 10  runtime       1530 non-null   object
 11  studio        494 non-null    object
dtypes: int64(1), object(11)
memory usage: 146.4+ KB


Very little box office ($) info.  Probably just use gross info from BOM dataframe.  Missing only 3 ratings, 8 genres, and 30 runtimes, so maybe just drop those rows.  Theater_date/dvd_date: we do have the years from the BOM at least.  Missing few hundred directors and writers, which could be problematic.  Have studios from BOM dataframe.  

In [17]:
rt_mi_df.rating.value_counts()

R        521
NR       503
PG       240
PG-13    235
G         57
NC17       1
Name: rating, dtype: int64

Ratings (for nonnull at least) look good. 

In [18]:
rt_mi_df.genre.value_counts().head(50).sort_index

<bound method Series.sort_index of Drama                                                      151
Comedy                                                     110
Comedy|Drama                                                80
Drama|Mystery and Suspense                                  67
Art House and International|Drama                           62
Action and Adventure|Drama                                  42
Action and Adventure|Drama|Mystery and Suspense             40
Drama|Romance                                               35
Comedy|Romance                                              32
Horror                                                      31
Art House and International|Comedy|Drama                    31
Action and Adventure|Science Fiction and Fantasy            24
Comedy|Drama|Romance                                        23
Classics|Drama                                              21
Action and Adventure|Mystery and Suspense                   20
Action and Adventure

OK this is ridiculous.  You have several different genres for the same movie, separated by "|".  Need to deal with this somehow.  Could create true-false Boolean columns like "is_drama", "is_comedy", etc  

In [19]:
rt_mi_df[rt_mi_df.director.str.contains("|", regex=False, na=False)]

id                                           synopsis rating  \
43      56  Albert Hughes and his brother Allen Hughes fol...      R   
53      67  After a mysterious, lost night on their honeym...      R   
100    128                                                NaN     NR   
121    155  Taking place over the course of one night, The...      R   
146    191  Award-winning filmmaker Rakeysh Omprakash Mehr...     NR   
...    ...                                                ...    ...   
1490  1918  A collection of 15 chapters from Tarzan and th...     NR   
1511  1945  Left on a nun's doorstep, Larry, Curly and Moe...     PG   
1518  1953  A glimpse into the comedic process and private...      R   
1529  1966  Two rabid Boston B-ball fans go over the edge ...  PG-13   
1544  1983  This animated children's film spoofs the Sherl...      G   

                                               genre  \
43                                             Drama   
53                                            Horror   
100                      Art House and International   
121                                     Comedy|Drama   
146                                            Drama   
...                                              ...   
1490       Action and Adventure|Classics|Documentary   
1511                                          Comedy   
1518                   Comedy|Documentary|Television   
1529                                          Comedy   
1544  Action and Adventure|Animation|Kids and Family   

                                               director  \
43                           Allen Hughes|Albert Hughes   
53                  Matt Bettinelli-Olpin|Tyler Gillett   
100                                    Ye Ying|Zhang Li   
121                   Jennifer Jason Leigh|Alan Cumming   
146     Rakeysh Omprakash Mehra|Rakeysh Omprakesh Mehra   
...                                                 ...   
1490                           Henry MacRae|Harry McRae   
1511                      Bobby Farrelly|Peter Farrelly   
1518                          Ricki Stern|Anne Sundberg   
1529                       Tom DeCerchio|Tom De Cerchio   
1544  Burny Mattinson|Ron Clements|John Musker|Dave ...   

                                              writer  theater_date  \
43    Michael Henry Brown|Albert Hughes|Allen Hughes   Oct 6, 1995   
53                                    Lindsay Devlin  Jan 17, 2014   
100                                              NaN           NaN   
121                Jennifer Jason Leigh|Alan Cumming   Jun 8, 2001   
146                                    Prasoon Joshi  Jul 12, 2013   
...                                              ...           ...   
1490         William Lord Wright|Ian McCloskey Heath   Oct 1, 1929   
1511      Bobby Farrelly|Peter Farrelly|Mike Cerrone  Apr 13, 2012   
1518                                     Ricki Stern  Jun 11, 2010   
1529                         Judd Apatow|Colin Quinn  Apr 19, 1996   
1544                                             NaN   Jul 2, 1986   

          dvd_date currency  box_office      runtime                  studio  
43    May 19, 1998      NaN         NaN  119 minutes                     NaN  
53    Apr 29, 2014      NaN         NaN   89 minutes        20th Century Fox  
100            NaN      NaN         NaN  120 minutes                     NaN  
121   Jan 15, 2002        $   2,974,308  115 minutes      Fine Line Features  
146   Jan 12, 2015        $   1,626,289  188 minutes  Reliance Entertainment  
...            ...      ...         ...          ...                     ...  
1490  Oct 24, 2006      NaN         NaN  290 minutes                     NaN  
1511  Jul 17, 2012        $  41,800,000   92 minutes        20th Century Fox  
1518  Dec 14, 2010        $   2,927,972   84 minutes               IFC Films  
1529   Sep 3, 2002      NaN         NaN   90 minutes                     NaN  
1544  Jul 23, 2002      NaN         NaN   74 minute

How to deal with multiple directors/writers?  Only 96 movies with mult directors...

In [20]:
rt_mi_df[rt_mi_df.writer.str.contains("|", regex=False, na=False)]

id                                           synopsis rating  \
1        3  New York City, not-too-distant-future: Eric Pa...      R   
3        6  Michael Douglas runs afoul of a treacherous su...      R   
7       13  Stewart Kane, an Irishman living in the Austra...      R   
13      20  Rachel (Britt Ekland) is an Amish girl who rej...  PG-13   
14      21  Old-school comic Rodney "I get no respect" Dan...      R   
...    ...                                                ...    ...   
1550  1990  A hard-working Sicilian heads for Switzerland ...      R   
1551  1991  Popular rap group Run DMC stars as a trio of p...      R   
1552  1992  The title character, played by John Turturro, ...      R   
1556  1997  The popular Saturday Night Live sketch was exp...     PG   
1558  1999  The Sandlot is a coming-of-age story about a g...     PG   

                                                genre            director  \
1                   Drama|Science Fiction and Fantasy    David Cronenberg   
3                          Drama|Mystery and Suspense      Barry Levinson   
7                                               Drama        Ray Lawrence   
13                              Classics|Comedy|Drama    William Friedkin   
14                                             Comedy       Peter Baldwin   
...                                               ...                 ...   
1550         Art House and International|Comedy|Drama      Franco Brusati   
1551                       Action and Adventure|Drama          Rick Rubin   
1552                                     Comedy|Drama           Joel Coen   
1556               Comedy|Science Fiction and Fantasy        Steve Barron   
1558  Comedy|Drama|Kids and Family|Sports and Fitness  David Mickey Evans   

                                                writer  theater_date  \
1                         David Cronenberg|Don DeLillo  Aug 17, 2012   
3                      Paul Attanasio|Michael Crichton   Dec 9, 1994   
7                     Raymond Carver|Beatrix Christian  Apr 27, 2006   
13         Sidney Michaels|Arnold Schulman|Norman Lear  Dec 22, 1968   
14                      Rodney Dangerfield|Harry Basil  Jan 31, 1997   
...                                                ...           ...   
1550         Nino Manfredi|Franco Brusati|Iaia Fiastri   Jun 1, 1974   
1551                            Rick Rubin|Ric Menello           NaN   
1552                              Joel Coen|Ethan Coen  Aug 21, 1991   
1556  Terry Turner|Tom Davis|Dan Aykroyd|Bonnie Turner  Jul 23, 1993   
1558                  David Mickey Evans|Robert Gunter   Apr 1, 1993   

          dvd_date currency box_office      runtime                  studio  
1      Jan 1, 2013        $    600,000  108 minutes       Entertainment One  
3     Aug 27, 1997      NaN        NaN  128 minutes                     NaN  
7      Oct 2, 2007        $    224,114  123 minutes  Sony Pictures Classics  
13    May 20, 2008      NaN        NaN   97 minutes                     NaN  
14    Sep 20, 1999      NaN        NaN  106 minutes                     NaN  
...            ...      ...        ...          ...                     ...  
1550   May 2, 2005      NaN        NaN  112 minutes                     NaN  
1551           NaN      NaN        NaN   92 minutes                     NaN  
1552  May 20, 2003      NaN        NaN  116 minutes                     NaN  
1556  Apr 17, 2001      NaN        NaN   88 minutes       Paramount Vantage  
1558  Jan 29, 2002      NaN        NaN  101 minutes                     NaN  

[499 rows x 12 columns]

But there are 499 movies with multiple writers!

Let's look at the Rotten Tomatoes review table:

In [21]:
rt_rev_df = pd.read_csv("zippedData/rt.reviews.tsv.gz", compression = 'gzip', sep = '\t', encoding = 'unicode_escape')

In [23]:
rt_rev_df.head(20)

id                                             review rating   fresh  \
0    3  A distinctly gallows take on contemporary fina...    3/5   fresh   
1    3  It's an allegory in search of a meaning that n...    NaN  rotten   
2    3  ... life lived in a bubble in financial dealin...    NaN   fresh   
3    3  Continuing along a line introduced in last yea...    NaN   fresh   
4    3             ... a perverse twist on neorealism...     NaN   fresh   
5    3  ... Cronenberg's Cosmopolis expresses somethin...    NaN   fresh   
6    3  Quickly grows repetitive and tiresome, meander...      C  rotten   
7    3  Cronenberg is not a director to be daunted by ...    2/5  rotten   
8    3  Cronenberg's cold, exacting precision and emot...    NaN   fresh   
9    3  Over and above its topical urgency or the bit ...    NaN   fresh   
10   3  Cosmopolis" is a hypnotic examination of our m...    NaN   fresh   
11   3  While not one of Cronenberg's stronger films, ...     B-   fresh   
12   3  Robert Pattinson works mighty hard to make Cos...    2/4  rotten   
13   3  The anger over the injustice of the financial ...      B   fresh   
14   3  For those who like their Cronenberg thick and ...    3/4   fresh   
15   3  For better or worse - often both - Cosmopolis ...    3/5   fresh   
16   3  For one of the smartest films I've seen in a w...    4/5   fresh   
17   3      Its major problem is that it's not cinematic.      B   fresh   
18   3  It's fascinating to watch Pattinson actually a...    2/4  rotten   
19   3  A black comedy as dry and deadpan as a bleache...    4/4   fresh   

             critic  top_critic                        publisher  \
0        PJ Nabarro           0                  Patrick Nabarro   
1    Annalee Newitz           0                          io9.com   
2      Sean Axmaker           0                 Stream on Demand   
3     Daniel Kasman           0                             MUBI   
4               NaN           0                     Cinema Scope   
5   Michelle Orange           0                 Capital New York   
6    Eric D. Snider           0                  EricDSnider.com   
7      Matt Kelemen           0               Las Vegas CityLife   
8      Sean Axmaker           0                    Parallax View   
9      Kong Rithdee           0                     Bangkok Post   
10     Tim Grierson           0                         Deadspin   
11     Emanuel Levy           0                  EmanuelLevy.Com   
12   Christian Toto           0                    Big Hollywood   
13     Robert Roten           0              Laramie Movie Scope   
14      Marty Mapes           0                      Movie Habit   
15        Adam Ross           0                   The Aristocrat   
16    Patrick Kolan           0                   Shotgun Cinema   
17  Dennis Schwartz           0        Ozus' World Movie Reviews   
18    Sean P. Means           0                Salt Lake Tribune   
19     John Beifuss           0  Commercial Appeal (Memphis, TN)   

                  date  
0    November 10, 2018  
1         May 23, 2018  
2      January 4, 2018  
3    November 16, 2017  
4     October 12, 2017  
5   September 11, 2017  
6        July 17, 2013  
7       April 21, 2013  
8       March 24, 2013  
9        March 4, 2013  
10   February 12, 2013  
11    February 3, 2013  
12    January 15, 2013  
13     January 7, 2013  
14    October 20, 2012  
15  September 27, 2012  
16  September 26, 2012  
17  September 25, 2012  
18  September 14, 2012  
19  September 10, 2012

Cool. So there are several reviews for each movie in the other RT df.  Can merge on id.  

In [93]:
rt_rev_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 54432 entries, 0 to 54431
Data columns (total 8 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   id          54432 non-null  int64 
 1   review      48869 non-null  object
 2   rating      40915 non-null  object
 3   fresh       54432 non-null  object
 4   critic      51710 non-null  object
 5   top_critic  54432 non-null  int64 
 6   publisher   54123 non-null  object
 7   date        54432 non-null  object
dtypes: int64(2), object(6)
memory usage: 3.3+ MB


In [95]:
rt_rev_df[rt_rev_df.review.isna()]

id review rating   fresh             critic  top_critic  \
162       3    NaN      B   fresh       Emanuel Levy           0   
169       5    NaN    3/5   fresh       Cole Smithey           0   
170       5    NaN    2/5  rotten      Chuck O'Leary           0   
171       5    NaN    3/5   fresh      Philip Martin           0   
172       5    NaN    3/5   fresh         Eric Melin           0   
...     ...    ...    ...     ...                ...         ...   
54393  1999    NaN    4/5   fresh     Scott Weinberg           0   
54428  2000    NaN    1/5  rotten  Michael Szymanski           0   
54429  2000    NaN    2/5  rotten       Emanuel Levy           0   
54430  2000    NaN  2.5/5  rotten   Christopher Null           0   
54431  2000    NaN    3/5   fresh    Nicolas Lacroix           0   

                       publisher                date  
162              EmanuelLevy.Com   December 25, 2012  
169              ColeSmithey.com    October 10, 2005  
170             Fantastica Daily     October 9, 2005  
171    Arkansas Democrat-Gazette      April 28, 2005  
172                 Lawrence.com     August 24, 2004  
...                          ...                 ...  
54393            eFilmCritic.com       July 25, 2002  
54428                 Zap2it.com  September 21, 2005  
54429            EmanuelLevy.Com       July 17, 2005  
54430             Filmcritic.com   September 7, 2003  
54431               Showbizz.net   November 12, 2002  

[5563 rows x 8 columns]

A lot of reviews are missing, but that's probably ok.  All of them have at least fresh or rotten.

In [98]:
rt_rev_df.rating.value_counts().head(50)

3/5       4327
4/5       3672
3/4       3577
2/5       3160
2/4       2712
2.5/4     2381
3.5/4     1777
3.5/5     1289
5/5       1237
B         1163
1/5       1113
1.5/4     1095
4/4        995
2.5/5      992
B+         832
1/4        822
B-         821
C          779
C+         665
4.5/5      567
7/10       522
A-         514
8/10       505
C-         493
6/10       468
1.5/5      405
A          397
5/10       351
D          324
9/10       304
4/10       252
D+         204
0/5        162
8          143
3/10       140
1          138
0/4        132
7          125
F          109
0.5/4      100
D-          84
6           82
2/10        73
A+          73
7.5/10      58
0.5/5       52
5           48
8.5/10      41
6.5/10      39
4           34
Name: rating, dtype: int64

The rating system is a little all over.  Need to make consistent.  Most are out of 5.  Some are out of 10.  Some are letter-graded.  Shouldn't be too too hard to fix, just .replace.

In [99]:
rt_rev_df.fresh.value_counts()

fresh     33035
rotten    21397
Name: fresh, dtype: int64

Perfect.  So can calculate a fresh percentage for each movie.  

In [100]:
rt_rev_df.top_critic.value_counts()

0    41336
1    13096
Name: top_critic, dtype: int64

What is top critic and do I care?  Maybe just look at top critics to get review data from more reputable sources.  

Also, there seems to be no way to tell what movies these are.  Can use to try to distinguish which genres, writers, directors have high rated movies and fresh movies.  

Now reading database from The Movie DB

In [25]:
tmdb_df = pd.read_csv("zippedData/tmdb.movies.csv.gz", compression = 'gzip', index_col=0)

In [26]:
tmdb_df.head(20)

genre_ids     id original_language  \
0            [12, 14, 10751]  12444                en   
1        [14, 12, 16, 10751]  10191                en   
2              [12, 28, 878]  10138                en   
3            [16, 35, 10751]    862                en   
4              [28, 878, 12]  27205                en   
5            [12, 14, 10751]  32657                en   
6          [28, 12, 14, 878]  19995                en   
7            [16, 10751, 35]  10193                en   
8            [16, 10751, 35]  20352                en   
9   [16, 28, 35, 10751, 878]  38055                en   
10           [16, 35, 10751]    863                en   
11           [10751, 14, 12]  12155                en   
12              [53, 12, 28]  27578                en   
13               [16, 10751]  38757                en   
14                  [27, 80]  41439                en   
15       [12, 14, 18, 10749]  24021                en   
16             [28, 53, 878]  20504                en   
17            [10402, 10749]  52371                nl   
18              [28, 18, 53]   2502                en   
19            [18, 53, 9648]  11324                en   

                                       original_title  popularity  \
0        Harry Potter and the Deathly Hallows: Part 1      33.533   
1                            How to Train Your Dragon      28.734   
2                                          Iron Man 2      28.515   
3                                           Toy Story      28.005   
4                                           Inception      27.920   
5   Percy Jackson & the Olympians: The Lightning T...      26.691   
6                                              Avatar      26.526   
7                                         Toy Story 3      24.445   
8                                       Despicable Me      23.673   
9                                            Megamind      22.855   
10                                        Toy Story 2      22.698   
11                                Alice in Wonderland      22.020   
12                                    The Expendables      21.517   
13                                            Tangled      21.511   
14                                             Saw 3D      20.370   
15                         The Twilight Saga: Eclipse      20.340   
16                                    The Book of Eli      18.985   
17                                         LelleBelle      18.586   
18                               The Bourne Supremacy      18.199   
19                                     Shutter Island      18.060   

   release_date                                              title  \
0    2010-11-19       Harry Potter and the Deathly Hallows: Part 1   
1    2010-03-26                           How to Train Your Dragon   
2    2010-05-07                                         Iron Man 2   
3    1995-11-22                                          Toy Story   
4    2010-07-16                                          Inception   
5    2010-02-11  Percy Jackson & the Olympians: The Lightning T...   
6    2009-12-18                                             Avatar   
7    2010-06-17                                        Toy Story 3   
8    2010-07-09                                      Despicable Me   
9    2010-11-04                                           Megamind   
10   1999-11-24                                        Toy Story 2   
11   2010-03-05                                Alice in Wonderland   
12   2010-08-03                                    The Expendables   
13   2010-11-24                                            Tangled   
14   2010-10-28                             Saw: The Final Chapter   
15   2010-06-23                         The Twilight Saga: Eclipse   
16   2010-01-11                                    The Book of Eli   
17   2010-10-09                                         LelleBelle   
18   2004-07-23                          

movies appear to have a user score out of 10.  We also have info on how many users voted.  Not sure what popularity means exactly.  No idea what to do with these genre ids.  Would be nice to have the genre ids key so I can look at which genres have higher ratings.  Basically just have titles and votes at this point.  hmm. 

In [110]:
tmdb_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 26517 entries, 0 to 26516
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   genre_ids          26517 non-null  object 
 1   id                 26517 non-null  int64  
 2   original_language  26517 non-null  object 
 3   original_title     26517 non-null  object 
 4   popularity         26517 non-null  float64
 5   release_date       26517 non-null  object 
 6   title              26517 non-null  object 
 7   vote_average       26517 non-null  float64
 8   vote_count         26517 non-null  int64  
dtypes: float64(2), int64(2), object(5)
memory usage: 2.0+ MB


No nulls?!?!?!?

Let's look at "The Numbers" data

In [112]:
tn_df = pd.read_csv("zippedData/tn.movie_budgets.csv.gz", compression= 'gzip')

In [113]:
tn_df.head()

id  release_date                                        movie  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

  production_budget domestic_gross worldwide_gross  
0      $425,000,000   $760,507,625  $2,776,345,279  
1      $410,600,000   $241,063,875  $1,045,663,875  
2      $350,000,000    $42,762,350    $149,762,350  
3      $330,600,000   $459,005,868  $1,403,013,963  
4      $317,000,000   $620,181,382  $1,316,721,747

So we have each movie, its production budget, and gross (domestic and worldwide).  Gross will probably be most useful.  Insights regarding whether investing outside the US might be interesting, but generally let's get info about which genres, writers, directors have the highest gross.  

In [114]:
tn_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   id                 5782 non-null   int64 
 1   release_date       5782 non-null   object
 2   movie              5782 non-null   object
 3   production_budget  5782 non-null   object
 4   domestic_gross     5782 non-null   object
 5   worldwide_gross    5782 non-null   object
dtypes: int64(1), object(5)
memory usage: 271.2+ KB


hooray!!! no nulls!

### SQL shizz

Let's import SQLite and establish a connection

In [115]:
import sqlite3

conn = sqlite3.Connection("im.db")

In [116]:
pd.read_sql("""
    SELECT *
    FROM persons

""",conn)

person_id         primary_name  birth_year  death_year  \
0       nm0061671    Mary Ellen Bauder         NaN         NaN   
1       nm0061865         Joseph Bauer         NaN         NaN   
2       nm0062070           Bruce Baum         NaN         NaN   
3       nm0062195         Axel Baumann         NaN         NaN   
4       nm0062798          Pete Baxter         NaN         NaN   
...           ...                  ...         ...         ...   
606643  nm9990381         Susan Grobes         NaN         NaN   
606644  nm9990690          Joo Yeon So         NaN         NaN   
606645  nm9991320       Madeline Smith         NaN         NaN   
606646  nm9991786  Michelle Modigliani         NaN         NaN   
606647  nm9993380       Pegasus Envoyé         NaN         NaN   

                                      primary_profession  
0              miscellaneous,production_manager,producer  
1             composer,music_department,sound_department  
2                             miscellaneous,actor,writer  
3       camera_department,cinematographer,art_department  
4       production_designer,art_department,set_decorator  
...                                                  ...  
606643                                           actress  
606644                                           actress  
606645                                           actress  
606646                                          producer  
606647                             director,actor,writer  

[606648 rows x 5 columns]

In [117]:
pd.read_sql("""
    SELECT *
    FROM writers

""",conn)

movie_id   person_id
0       tt0285252   nm0899854
1       tt0438973   nm0175726
2       tt0438973   nm1802864
3       tt0462036   nm1940585
4       tt0835418   nm0310087
...           ...         ...
255868  tt8999892  nm10122246
255869  tt8999974  nm10122357
255870  tt9001390   nm6711477
255871  tt9004986   nm4993825
255872  tt9010172   nm8352242

[255873 rows x 2 columns]

Cool, so can merge with persons on person_id to get person info, and use movie_id to get movie info, and eventually gross info. 

In [118]:
pd.read_sql("""
    SELECT *
    FROM directors
""",conn)

movie_id   person_id
0       tt0285252   nm0899854
1       tt0462036   nm1940585
2       tt0835418   nm0151540
3       tt0835418   nm0151540
4       tt0878654   nm0089502
...           ...         ...
291169  tt8999974  nm10122357
291170  tt9001390   nm6711477
291171  tt9001494  nm10123242
291172  tt9001494  nm10123248
291173  tt9004986   nm4993825

[291174 rows x 2 columns]

similar

In [119]:
pd.read_sql("""
    SELECT *
    FROM known_for
""",conn)

person_id   movie_id
0        nm0061671  tt0837562
1        nm0061671  tt2398241
2        nm0061671  tt0844471
3        nm0061671  tt0118553
4        nm0061865  tt0896534
...            ...        ...
1638255  nm9990690  tt9090932
1638256  nm9990690  tt8737130
1638257  nm9991320  tt8734436
1638258  nm9991320  tt9615610
1638259  nm9993380  tt8743182

[1638260 rows x 2 columns]

Not sure what to make of this. 

In [121]:
pd.read_sql("""
    SELECT *
    FROM principals
""",conn)

movie_id  ordering   person_id  category       job  \
0        tt0111414         1   nm0246005     actor      None   
1        tt0111414         2   nm0398271  director      None   
2        tt0111414         3   nm3739909  producer  producer   
3        tt0323808        10   nm0059247    editor      None   
4        tt0323808         1   nm3579312   actress      None   
...            ...       ...         ...       ...       ...   
1028181  tt9692684         1   nm0186469     actor      None   
1028182  tt9692684         2   nm4929530      self      None   
1028183  tt9692684         3  nm10441594  director      None   
1028184  tt9692684         4   nm6009913    writer    writer   
1028185  tt9692684         5  nm10441595  producer  producer   

                   characters  
0                 ["The Man"]  
1                        None  
2                        None  
3                        None  
4            ["Beth Boothby"]  
...                       ...  
1028181  ["Ebenezer Scrooge"]  
1028182   ["Herself","Regan"]  
1028183                  None  
1028184                  None  
1028185                  None  

[1028186 rows x 6 columns]

Not sure how useful this is.

In [123]:
pd.read_sql("""
    SELECT *
    FROM movie_basics
""",conn)

movie_id                                primary_title  \
0       tt0063540                                    Sunghursh   
1       tt0066787              One Day Before the Rainy Season   
2       tt0069049                   The Other Side of the Wind   
3       tt0069204                              Sabse Bada Sukh   
4       tt0100275                     The Wandering Soap Opera   
...           ...                                          ...   
146139  tt9916538                          Kuambil Lagi Hatiku   
146140  tt9916622  Rodolpho Teóphilo - O Legado de um Pioneiro   
146141  tt9916706                              Dankyavar Danka   
146142  tt9916730                                       6 Gunn   
146143  tt9916754               Chico Albuquerque - Revelações   

                                     original_title  start_year  \
0                                         Sunghursh        2013   
1                                   Ashad Ka Ek Din        2019   
2                        The Other Side of the Wind        2018   
3                                   Sabse Bada Sukh        2018   
4                             La Telenovela Errante        2017   
...                                             ...         ...   
146139                          Kuambil Lagi Hatiku        2019   
146140  Rodolpho Teóphilo - O Legado de um Pioneiro        2015   
146141                              Dankyavar Danka        2013   
146142                                       6 Gunn        2017   
146143               Chico Albuquerque - Revelações        2013   

        runtime_minutes                genres  
0                 175.0    Action,Crime,Drama  
1                 114.0       Biography,Drama  
2                 122.0                 Drama  
3                   NaN          Comedy,Drama  
4                  80.0  Comedy,Drama,Fantasy  
...                 ...                   ...  
146139            123.0                 Drama  
146140              NaN           Documentary  
146141              NaN                Comedy  
146142            116.0                  None  
146143              NaN           Documentary  

[146144 rows x 6 columns]

So we have movie_ids--can grab writers, directors.  But also have titles.  Can use with gross info (bom or tn) as well as ratings info (hooray, i can actually use the ratings from tmdb).  

In [124]:
pd.read_sql("""
    SELECT *
    FROM movie_ratings
""",conn)

movie_id  averagerating  numvotes
0      tt10356526            8.3        31
1      tt10384606            8.9       559
2       tt1042974            6.4        20
3       tt1043726            4.2     50352
4       tt1060240            6.5        21
...           ...            ...       ...
73851   tt9805820            8.1        25
73852   tt9844256            7.5        24
73853   tt9851050            4.7        14
73854   tt9886934            7.0         5
73855   tt9894098            6.3       128

[73856 rows x 3 columns]

Also have movie ratings here.

In [125]:
pd.read_sql("""
    SELECT *
    FROM movie_akas
""",conn)

movie_id  ordering                                    title region  \
0       tt0369610        10                            Джурасик свят     BG   
1       tt0369610        11                        Jurashikku warudo     JP   
2       tt0369610        12  Jurassic World: O Mundo dos Dinossauros     BR   
3       tt0369610        13                  O Mundo dos Dinossauros     BR   
4       tt0369610        14                           Jurassic World     FR   
...           ...       ...                                      ...    ...   
331698  tt9827784         2                       Sayonara kuchibiru   None   
331699  tt9827784         3                            Farewell Song    XWW   
331700  tt9880178         1                              La atención   None   
331701  tt9880178         2                              La atención     ES   
331702  tt9880178         3                            The Attention    XWW   

       language        types   attributes  is_original_title  
0            bg         None         None                0.0  
1          None  imdbDisplay         None                0.0  
2          None  imdbDisplay         None                0.0  
3          None         None  short title                0.0  
4          None  imdbDisplay         None                0.0  
...         ...          ...          ...                ...  
331698     None     original         None                1.0  
331699       en  imdbDisplay         None                0.0  
331700     None     original         None                1.0  
331701     None         None         None                0.0  
331702       en  imdbDisplay         None                0.0  

[331703 rows x 8 columns]

Might not need this.